processing/src/main/java/org/apache/druid/segment/AutoTypeColumnIndexer.java

```diff
@@ -557,7 +613,14 @@ public FieldTypeInfo.MutableTypeSet getTypes()
 
     public boolean isSingleType()
     {
-      return typeSet.getSingleType() != null;
+      ColumnType singleType = typeSet.getSingleType();
+      if (singleType != null) {
+        if (typeSet.hasEmptyArray()) {
```

Comment:
I looked at FieldTypeInfo and convinced myself that this should be checked when `singleType` is null, not when it's not null.  Did I just confuse myself?

Reply:
oops yeah this is stale, i wrote this part before i pushed checking empty array bit into `getSingleType`